offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -802,6 +802,223 @@ export class CkbTxGenerator extends CkbTxHelper {
+      hash_type: ForceBridgeCore.config.ckb.deps.sudtType.script.hashType,
+      args: sudtArgs,
+    };
+
+    let totalSudtOutputCapacity = BigInt(0);
+    let totalSudtOutputAmount = BigInt(0);
+    // add recipient outputs
+    records.forEach((record) => {
+      const amount = BigInt(record.amount);
+      const recipientCell: Cell = {
+        cell_output: {
+          capacity: '0x0',
+          lock: parseAddress(record.recipient),
+          type: sudtScript,
+        },
+        data: utils.toBigUInt128LE(amount),
+      };
+      const recipientCellMinimalCapacity = minimalCellCapacity(recipientCell);
+      totalSudtOutputCapacity += recipientCellMinimalCapacity;
+      totalSudtOutputAmount += amount;
+      recipientCell.cell_output.capacity = `0x${recipientCellMinimalCapacity.toString(16)}`;
+      txSkeleton = txSkeleton.update('outputs', (outputs) => {
+        return outputs.push(recipientCell);
+      });
+    });
+
+    // collect sudt and add multisig cell inputs
+    const omniLockMultisigScript = parseAddress(getOmniLockMultisigAddress());
+    const omniLockMultisigChangeCell: Cell = {
+      cell_output: {
+        capacity: `0x0`,
+        lock: omniLockMultisigScript,
+        type: sudtScript,
+      },
+      data: '0x',
+    };
+    const searchKey = {
+      script: omniLockMultisigScript,
+      script_type: ScriptType.lock,
+      filter: {
+        script: {
+          code_hash: ForceBridgeCore.config.ckb.deps.sudtType.script.codeHash,
+          hash_type: ForceBridgeCore.config.ckb.deps.sudtType.script.hashType,
+          args: sudtArgs,
+        },
+      },
+    };
+    const searchedMultisigCells = await this.collector.collectSudtByAmount(searchKey, totalSudtOutputAmount);
+    const searchedMultisigCellsAmount = searchedMultisigCells
+      .map((cell) => utils.readBigUInt128LE(cell.data))
+      .reduce((a, b) => a + b, 0n);
+    if (searchedMultisigCellsAmount < totalSudtOutputAmount)
+      throw new Error(
+        `multisig cell sudt amount not enough, need ${totalSudtOutputAmount.toString()}, found ${searchedMultisigCellsAmount.toString()}`,
+      );
+    txSkeleton = txSkeleton.update('inputs', (inputs) => {
+      return inputs.push(...searchedMultisigCells);
+    });
+    const searchedMultisigCellsCapacity = searchedMultisigCells
+      .map((cell) => BigInt(cell.cell_output.capacity))
+      .reduce((a, b) => a + b, 0n);
+
+    // add multisigChangeCell if needed
+    let omniLockMultisigChangeCellMinimalCapacity = BigInt(0);
+    if (searchedMultisigCellsAmount > totalSudtOutputAmount) {
+      const multisigChangeCellAmount = searchedMultisigCellsAmount - totalSudtOutputAmount;
+      omniLockMultisigChangeCell.data = `0x${multisigChangeCellAmount.toString(16)}`;
+      omniLockMultisigChangeCellMinimalCapacity = minimalCellCapacity(omniLockMultisigChangeCell);
+      omniLockMultisigChangeCell.cell_output.capacity = `0x${omniLockMultisigChangeCellMinimalCapacity.toString(16)}`;
```

Comment:
omniLockMultisigChangeCell.cell_output.capacity = multisig_cell_1_capacity_in_input +  multisig_cell_2_capacity_in_input + ...;

In other words, we recycle all the capacity from input multsigCells, and we use collector account to supply capacity to recipient cells and to tx fee.

```
if (searchedMultisigCellsAmount > totalSudtOutputAmount) {
  // all the capacity from input multsigCells are recycled in omniLockMultisigSudtChangeCell
}

if (searchedMultisigCellsAmount === totalSudtOutputAmount) {
  // all the capacity from input multsigCells are recycled in omniLockMultisigCKBChangeCell
}
```

Reply:
if searchedMultisigCellsAmount === totalSudtOutputAmount then use ckb cell for change